src/components/app/MenuButtons/Permalink.js

```diff
@@ -28,15 +28,30 @@ type State = {|
 
 export class MenuButtonsPermalink extends React.PureComponent<Props, State> {
   _permalinkTextField: HTMLInputElement | null = null;
+  _buttonWithPanel: ButtonWithPanel | null = null;
```

Comment:
Looking at this, I think I would like to avoid taking another `ref` if possible.

Currently it does the job, but the same functionality can be implemented by a prop. Actually we already have this prop as `initialOpen`. So all we have to do is to make this prop more generic, like `open` and make sure that it's being handled properly on `componentDidUpdate` inside that component.

I checked other places and it looks like here is the only place where we use `initialOpen`, so it should be fine to do that.

It's usually fix things with props if we can instead of refs because that way we let the component handle its state internally. And we make sure that this component doesn't have a different behavior by not breaking the proper encapsulation.